core/src/cloud/grow/auth.ts

```diff
@@ -86,8 +87,15 @@ export async function refreshAuthTokenAndWriteToConfigStore(
     }
 
     log.debug({ msg: `Failed to refresh the token.` })
+    if (err.message.toLowerCase().includes("invalid refresh token")) {
```

Comment:
Can we also determine the token validity using the response status code?
```suggestion
    if (err.statusCode >= 400 && err.statusCode < 500) {
```

Reply:
Maybe we should even test for 401 explicitly

Reply:
If we ever migrate to other kinds of tokens that need to be decoded, e.g. JWT tokens then we should also delete the token on 400 response though, so I think `400 <= code < 500` is fine

Reply:
Backend V2 did not return any status code for invalid token.

Reply:
We need to change the backend in that case, to make sure that it will emit correct status code in this case then. Maybe @hph can support you on this?

Reply:
The error returned is `UNAUTHORIZED`, which should map to 401. This status code is available in the `httpStatus` property, at least when using the frontend client, but I think it should be the same in core. [Reference (see error shape above the linked table)](https://trpc.io/docs/server/error-handling#error-codes)

Re. the error itself, currently this can only happen if a) the token doesn't exist in the database or b) the token is expired.

Reply:
I double-check that and did not see the status code in the `TRPCClientError` instance. The `data` property (that should contain `httpStatus`) was `undefined`, I'm not sure why.

Reply:
I found the original error code deep inside the error object.

Reply:
Is it maybe helpful to infer the specific error shape for the router? See also https://trpc.io/docs/client/vanilla/infer-types#infer-trpcclienterror-types

Reply:
We removed unnecessary error transforming from the TRPC client config in #6964. So, now we can use original httpStatus from a `TRPCClientError` instance.